onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc

```diff
@@ -5,42 +5,51 @@
 #pragma warning(disable : 4996)
 #endif
 
-#include "core/framework/allocator.h"
-#include "core/framework/compute_capability.h"
-#include "core/framework/kernel_registry.h"
-#include "core/providers/dnnl/subgraph/dnnl_func_kernel.h"
+#include "core/providers/shared_library/provider_author.h"
+#include <unordered_set>
+#include "subgraph/dnnl_func_kernel.h"
 #include "dnnl_execution_provider.h"
 #include "dnnl_fwd.h"
 
-namespace {
+const OrtApi* ORT_API_CALL GetApi(uint32_t /*version*/) NO_EXCEPTION { return nullptr; }
```

Comment:
Does every execution provider that wants to be a DLL need to do all this? Seems like boiler plate.

Reply:
I found a way to not include the C++ header in two places and use the macro to disable initializing that variable